lib/base/io-engine.cpp

```diff
@@ -85,6 +85,11 @@ boost::asio::io_context& IoEngine::GetIoContext()
 	return m_IoContext;
 }
 
+int_fast32_t IoEngine::GetCpuSemaphore() const
+{
+	return m_CpuBoundSemaphore;
```

Comment:
@julianbrost Also, I'm wondering whether we can do this:

```suggestion
	return m_CpuBoundSemaphore.load(std::memory_order_relaxed);
```

Reply:
That function isn't used to synchronize access to any other memory, so ordering doesn't matter here (i.e. everything[^1] would be fine).

[^1]: Of course except those that result in undefined behavior when used for a load operation, C++ is fun.